client-app/src/pages/account/components/payments-filter/payments-filter.ts

```diff
@@ -3,70 +3,63 @@ import Component from "vue-class-component";
 import { LocaleMessages } from "vue-i18n";
 import { Prop } from "vue-property-decorator";
 import i18n from "@i18n";
-import { PaymentsListFilters } from "@account/store/modules/payments-list/types";
+import { IPaymentSearchCriteria } from '@common/api/api-clients';
 import { locale } from "@common/constants";
 
 @Component
 export default class PaymentsFilter extends Vue {
   @Prop()
-  paymentsFilter!: PaymentsListFilters;
-
-  startDate: Date | undefined = undefined;
-
-  endDate: Date | undefined = undefined;
+  searchCriteria!: IPaymentSearchCriteria;
 
   isDateValid: boolean | null = null;
 
   datepickerLabels: LocaleMessages | {} = {};
 
   locale = locale;
 
-  emitChanges(updatedFilters: PaymentsListFilters) {
-    this.$emit("filtersChanged", updatedFilters);
+  emitChanges(searchCriteria: IPaymentSearchCriteria) {
+    this.$emit("searchCriteriaChanged", searchCriteria);
   }
 
   mounted() {
     this.setDatepickerLocalization();
   }
 
-  dateChanged() {
-    if (this.startDate && this.endDate) {
-      this.isDateValid = this.startDate <= this.endDate;
+  dateChanged(startDate?: Date, endDate?: Date) {
+    if (startDate && endDate) {
+      this.isDateValid = startDate <= endDate;
       if (this.isDateValid) {
-        const updatedFilters = { ...this.paymentsFilter, startDate: this.startDate, endDate: this.prepareEndDate() };
-        this.emitChanges(updatedFilters);
+        const searchCriteria = { ...this.searchCriteria, startDate, endDate };
+        this.emitChanges(searchCriteria);
       }
```

Comment:
Are you sure that it is ok when are we not emitting the event for an invalid date? ) Here we have really changed date (if look at caller code), so we have to emit any changes anyway cuz our func called dateChanged

Reply:
Same as above